app-android/app/src/main/java/de/tutao/tutanota/credentials/CredentialsEncryptionBeforeAPI30.kt

```diff
@@ -1,109 +1,101 @@
 package de.tutao.tutanota.credentials
 
+import android.content.Context
 import android.security.keystore.KeyPermanentlyInvalidatedException
 import android.security.keystore.UserNotAuthenticatedException
 import androidx.biometric.BiometricManager
 import androidx.biometric.BiometricPrompt
 import androidx.biometric.BiometricPrompt.PromptInfo
 import androidx.fragment.app.FragmentActivity
 import de.tutao.tutanota.AndroidKeyStoreFacade
+import de.tutao.tutanota.AndroidNativeCryptoFacade
 import de.tutao.tutanota.CredentialAuthenticationException
 import de.tutao.tutanota.CryptoError
 import de.tutao.tutanota.R
-import de.tutao.tutanota.ipc.DataWrapper
-import de.tutao.tutanota.ipc.NativeCredentialsFacade
-import de.tutao.tutanota.ipc.wrap
+import de.tutao.tutanota.ipc.CredentialEncryptionMode
 import java.security.KeyStoreException
 import javax.crypto.Cipher
 
 class CredentialsEncryptionBeforeAPI30(
-		private val keyStoreFacade: AndroidKeyStoreFacade,
-		private val activity: FragmentActivity,
-		private val authenticationPrompt: AuthenticationPrompt,
-) : NativeCredentialsFacade {
+	private val keyStoreFacade: AndroidKeyStoreFacade,
+	private val activity: Context,
```

Comment:
Because this is also used from a `JobService` alongside the `MainActivity`. The `as FragmentActivity` ones are not called from the service. We can maybe refactor this later, for now it needs to use `Context` to match both.